app/scripts/contentscript.js

```diff
@@ -103,6 +104,23 @@ const IGNORE_INIT_METHODS_FOR_KEEP_ALIVE = [
 let keepAliveInterval;
 let keepAliveTimer;
 
+/**
+ * Sending a message to the extension to receive will keep the service worker alive.
+ *
+ * If the extension is unloaded or reloaded during a session and the user attempts to send a
+ * message to the extension, an "Extension context invalidated." error will occur.
+ * When this happens, prompt the user to reload the extension.
+ */
+const sendMessageWorkerKeepAlive = () => {
+  browser.runtime
+    .sendMessage({ name: WORKER_KEEP_ALIVE_MESSAGE })
+    .then(null, (error) => {
```

Comment:
maybe easier to read as a `.catch`

Reply:
I was following the semantics the [sendMessage doc uses](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage#examples). I think using `.catch` is easier to read as well. I plan to update this some time after I complete some MV3 work